File: bot/cogs/economy.py
import discord
from discord.ext import commands
from discord import app_commands
from datetime import datetime, timedelta
import random
from typing import Optional
from database.connection import AsyncSessionLocal
from database.models import Member, Economy
from bot.utils.embeds import success_embed, error_embed, info_embed
from sqlalchemy import select
from sqlalchemy.orm import selectinload


class EconomyCog(commands.Cog):
    def __init__(self, bot):
        self.bot = bot
        self.shop_items = {
            "vip": {"name": "VIP Pass", "price": 10000, "description": "VIP role for 30 days", "emoji": "⭐"},
            "color": {"name": "Custom Color", "price": 5000, "description": "Role with a custom color", "emoji": "🎨"},
            "boost": {"name": "XP Boost x2", "price": 3000, "description": "Double XP for 24 hours", "emoji": "⚡"},
            "coins": {"name": "Coin Pack", "price": 1000, "description": "Bonus 1,000 coins", "emoji": "💰"},
        }

    async def get_or_create_member(self, session, guild_id: int, user_id: int, username: str):
        result = await session.execute(
            select(Member).options(selectinload(Member.economy)).where(
                Member.guild_id == guild_id,
                Member.user_id == user_id
            )
        )
        member = result.scalar_one_or_none()

        if not member:
            member = Member(
                guild_id=guild_id,
                user_id=user_id,
                username=username,
                coins=0,
                bank=0
            )
            session.add(member)
            await session.flush()

            economy = Economy(member_id=member.id, inventory={})
            session.add(economy)
            await session.flush()
            member.economy = economy

        if not member.economy:
            economy = Economy(member_id=member.id, inventory={})
            session.add(economy)
            await session.flush()
            member.economy = economy

        return member

    @commands.hybrid_command(name="balance", description="Show your balance or another member's")
    @commands.cooldown(1, 5, commands.BucketType.user)
    async def balance(self, ctx: commands.Context, member: discord.Member = None):
        member = member or ctx.author

        async with AsyncSessionLocal() as session:
            db_member = await self.get_or_create_member(session, ctx.guild.id, member.id, str(member))
            await session.commit()

            embed = discord.Embed(
                title=f"💰 Balance for {member.display_name}",
                color=discord.Color.gold(),
                timestamp=datetime.utcnow()
            )
            embed.add_field(name="💵 Wallet", value=f"{db_member.coins:,}", inline=True)
            embed.add_field(name="🏦 Bank", value=f"{db_member.bank:,}", inline=True)
            embed.add_field(name="💎 Total", value=f"{db_member.coins + db_member.bank:,}", inline=True)
            embed.set_thumbnail(url=member.display_avatar.url)
            await ctx.send(embed=embed)

    @commands.hybrid_command(name="work", description="Work to earn coins")
    @commands.cooldown(1, 3600, commands.BucketType.user)
    async def work(self, ctx: commands.Context):
        async with AsyncSessionLocal() as session:
            member = await self.get_or_create_member(session, ctx.guild.id, ctx.author.id, str(ctx.author))

            if member.last_work and datetime.utcnow() - member.last_work < timedelta(hours=1):
                remaining = timedelta(hours=1) - (datetime.utcnow() - member.last_work)
                minutes = int(remaining.total_seconds() // 60)
                await ctx.send(embed=error_embed(f"You already worked recently. Try again in {minutes} minute(s)"))
                return

            jobs = [
                "developer", "doctor", "lawyer", "teacher", "engineer",
                "artist", "musician", "chef", "architect", "scientist"
            ]
            job = random.choice(jobs)
            earned = random.randint(100, 500)

            member.coins += earned
            member.last_work = datetime.utcnow()

            if member.economy:
                member.economy.total_earned += earned
                member.economy.work_streak += 1

            await session.commit()

            embed = success_embed(f"You worked as **{job}** and earned **{earned}** coins")
            embed.set_footer(text=f"Work streak: {member.economy.work_streak if member.economy else 0}")
            await ctx.send(embed=embed)

    @commands.hybrid_command(name="daily", description="Claim your daily reward")
    @commands.cooldown(1, 86400, commands.BucketType.user)
    async def daily(self, ctx: commands.Context):
        async with AsyncSessionLocal() as session:
            member = await self.get_or_create_member(session, ctx.guild.id, ctx.author.id, str(ctx.author))

            if member.last_daily and datetime.utcnow() - member.last_daily < timedelta(days=1):
                remaining = timedelta(days=1) - (datetime.utcnow() - member.last_daily)
                hours = int(remaining.total_seconds() // 3600)
                await ctx.send(embed=error_embed(f"You already claimed your daily reward. Come back in {hours} hour(s)"))
                return

            base_reward = 500
            streak_bonus = 0

            if member.economy:
                if member.last_daily and datetime.utcnow() - member.last_daily <= timedelta(days=2):
                    member.economy.daily_streak += 1
                else:
                    member.economy.daily_streak = 1

                streak_bonus = min(member.economy.daily_streak * 50, 500)

            total_reward = base_reward + streak_bonus
            member.coins += total_reward
            member.last_daily = datetime.utcnow()

            if member.economy:
                member.economy.total_earned += total_reward

            await session.commit()

            embed = success_embed(f"You received your daily reward of **{total_reward}** coins")
            if streak_bonus > 0:
                embed.add_field(
                    name="🔥 Streak",
                    value=f"{member.economy.daily_streak} day(s) - Bonus: {streak_bonus} coins",
                    inline=False
                )
            await ctx.send(embed=embed)

    @commands.hybrid_command(name="shop", description="Show the shop")
    @commands.cooldown(1, 5, commands.BucketType.user)
    async def shop(self, ctx: commands.Context):
        embed = discord.Embed(
            title="🛒 Shop",
            description="Use `/buy <item>` to purchase an item",
            color=discord.Color.blue(),
            timestamp=datetime.utcnow()
        )

        for item_id, item_data in self.shop_items.items():
            embed.add_field(
                name=f"{item_data['emoji']} {item_data['name']}",
                value=f"{item_data['description']}\n**Price:** {item_data['price']:,} 💰\n**ID:** `{item_id}`",
                inline=False
            )

        await ctx.send(embed=embed)

    @commands.hybrid_command(name="buy", description="Purchase an item from the shop")
    @commands.cooldown(1, 5, commands.BucketType.user)
    async def buy(self, ctx: commands.Context, item: str):
        if item not in self.shop_items:
            await ctx.send(embed=error_embed("Item not found in the shop"))
            return

        item_data = self.shop_items[item]

        async with AsyncSessionLocal() as session:
            member = await self.get_or_create_member(session, ctx.guild.id, ctx.author.id, str(ctx.author))

            if member.coins < item_data["price"]:
                await ctx.send(embed=error_embed(f"You do not have enough coins. Cost: {item_data['price']:,} 💰"))
                return

            member.coins -= item_data["price"]

            if member.economy:
                member.economy.total_spent += item_data["price"]
                if not member.economy.inventory:
                    member.economy.inventory = {}
                inventory = member.economy.inventory
                inventory[item] = inventory.get(item, 0) + 1
                member.economy.inventory = inventory

            await session.commit()

            embed = success_embed(f"You purchased **{item_data['name']}** {item_data['emoji']}")
            embed.add_field(name="Price", value=f"{item_data['price']:,} 💰", inline=True)
            embed.add_field(name="Remaining balance", value=f"{member.coins:,} 💰", inline=True)
            await ctx.send(embed=embed)

    @commands.hybrid_command(name="inventory", description="Show your inventory")
    @commands.cooldown(1, 5, commands.BucketType.user)
    async def inventory(self, ctx: commands.Context, member: discord.Member = None):
        member = member or ctx.author

        async with AsyncSessionLocal() as session:
            db_member = await self.get_or_create_member(session, ctx.guild.id, member.id, str(member))
            await session.commit()

            embed = discord.Embed(
                title=f"🎒 Inventory for {member.display_name}",
                color=discord.Color.purple(),
                timestamp=datetime.utcnow()
            )

            if not db_member.economy or not db_member.economy.inventory or len(db_member.economy.inventory) == 0:
                embed.description = "Inventory empty"
            else:
                for item_id, quantity in db_member.economy.inventory.items():
                    if item_id in self.shop_items:
                        item_data = self.shop_items[item_id]
                        embed.add_field(
                            name=f"{item_data['emoji']} {item_data['name']}",
                            value=f"Quantity: {quantity}",
                            inline=True
                        )

            embed.set_thumbnail(url=member.display_avatar.url)
            await ctx.send(embed=embed)

    @commands.hybrid_command(name="transfer", description="Transfer coins to another member")
    @commands.cooldown(1, 10, commands.BucketType.user)
    async def transfer(self, ctx: commands.Context, member: discord.Member, amount: int):
        if member.bot:
            await ctx.send(embed=error_embed("You cannot transfer coins to a bot"))
            return

        if member.id == ctx.author.id:
            await ctx.send(embed=error_embed("You cannot transfer coins to yourself"))
            return

        if amount <= 0:
            await ctx.send(embed=error_embed("Amount must be greater than 0"))
            return

        async with AsyncSessionLocal() as session:
            sender = await self.get_or_create_member(session, ctx.guild.id, ctx.author.id, str(ctx.author))
            receiver = await self.get_or_create_member(session, ctx.guild.id, member.id, str(member))

            if sender.coins < amount:
                await ctx.send(embed=error_embed(f"You do not have enough coins. Balance: {sender.coins:,} 💰"))
                return

            sender.coins -= amount
            receiver.coins += amount

            await session.commit()

            embed = success_embed(f"You transferred **{amount:,}** coins to {member.mention}")
            embed.add_field(name="Your new balance", value=f"{sender.coins:,} 💰", inline=True)
            await ctx.send(embed=embed)

    @commands.hybrid_command(name="gamble", description="Gamble coins (50% chance to double)")
    @commands.cooldown(1, 30, commands.BucketType.user)
    async def gamble(self, ctx: commands.Context, amount: int):
        if amount <= 0:
            await ctx.send(embed=error_embed("Amount must be greater than 0"))
            return

        async with AsyncSessionLocal() as session:
            member = await self.get_or_create_member(session, ctx.guild.id, ctx.author.id, str(ctx.author))

            if member.coins < amount:
                await ctx.send(embed=error_embed(f"You do not have enough coins. Balance: {member.coins:,} 💰"))
                return

            win = random.choice([True, False])

            if win:
                member.coins += amount
                await session.commit()
                embed = discord.Embed(
                    title="🎰 You won!",
                    description=f"You gambled **{amount:,}** coins and won **{amount:,}** coins",
                    color=discord.Color.green(),
                    timestamp=datetime.utcnow()
                )
                embed.add_field(name="New balance", value=f"{member.coins:,} 💰", inline=True)
            else:
                member.coins -= amount
                await session.commit()
                embed = discord.Embed(
                    title="🎰 You lost!",
                    description=f"You gambled **{amount:,}** coins and lost everything",
                    color=discord.Color.red(),
                    timestamp=datetime.utcnow()
                )
                embed.add_field(name="New balance", value=f"{member.coins:,} 💰", inline=True)

            await ctx.send(embed=embed)


async def setup(bot):
    await bot.add_cog(EconomyCog(bot))

File: src/commands/economy.rs
use crate::database;
use crate::handler::{Context, Error};
use crate::utils::embeds::*;
use crate::utils::format_number;
use poise::serenity_prelude as serenity;
use rand::Rng;
use serde_json::json;

#[poise::command(slash_command, category = "Economy")]
pub async fn balance(
    ctx: Context<'_>,
    #[description = "User to check balance"] user: Option<serenity::User>,
) -> Result<(), Error> {
    let user = user.as_ref().unwrap_or_else(|| ctx.author());
    let guild_id = ctx.guild_id().ok_or("This command can only be used in a guild")?;

    let guild_id_db = guild_id.get() as i64;
    let user_id_db = user.id.get() as i64;

    let member = database::get_or_create_member(&ctx.data().db_pool, guild_id_db, user_id_db, &user.name).await?;

    let embed = serenity::CreateEmbed::default()
        .title(format!("💰 Balance for {}", user.name))
        .thumbnail(user.face())
        .field("💵 Wallet", format_number(member.coins as i64), true)
        .field("🏦 Bank", format_number(member.bank as i64), true)
        .field("💎 Total", format_number((member.coins + member.bank) as i64), true)
        .color(serenity::Color::from_rgb(255, 215, 0));

    ctx.send(poise::CreateReply::default().embed(embed)).await?;
    Ok(())
}

#[poise::command(slash_command, category = "Economy")]
pub async fn work(ctx: Context<'_>) -> Result<(), Error> {
    let guild_id = ctx.guild_id().ok_or("This command can only be used in a guild")?;
    let guild_id_db = guild_id.get() as i64;
    let user_id_db = ctx.author().id.get() as i64;

    let member = database::get_or_create_member(&ctx.data().db_pool, guild_id_db, user_id_db, &ctx.author().name).await?;

    let jobs = ["developer", "doctor", "lawyer", "teacher", "engineer", "artist", "musician", "chef", "architect", "scientist"];
    let (job, earned) = {
        let mut rng = rand::thread_rng();
        let job = jobs[rng.gen_range(0..jobs.len())];
        let earned = rng.gen_range(100..=500);
        (job, earned)
    };

    sqlx::query!(
        "UPDATE members SET coins = coins + ?, last_work = CURRENT_TIMESTAMP WHERE id = ?",
        earned,
        member.id
    )
    .execute(&ctx.data().db_pool)
    .await?;

    sqlx::query!(
        "UPDATE economy SET total_earned = total_earned + ?, work_streak = work_streak + 1 WHERE member_id = ?",
        earned,
        member.id
    )
    .execute(&ctx.data().db_pool)
    .await?;

    let embed = success_embed(&format!("You worked as **{}** and earned **{}** coins", job, earned));
    ctx.send(poise::CreateReply::default().embed(embed)).await?;
    Ok(())
}

#[poise::command(slash_command, category = "Economy")]
pub async fn daily(ctx: Context<'_>) -> Result<(), Error> {
    let guild_id = ctx.guild_id().ok_or("This command can only be used in a guild")?;
    let guild_id_db = guild_id.get() as i64;
    let user_id_db = ctx.author().id.get() as i64;

    let member = database::get_or_create_member(&ctx.data().db_pool, guild_id_db, user_id_db, &ctx.author().name).await?;

    let base_reward = 500;
    let mut streak_bonus = 0;

    let economy = sqlx::query!("SELECT daily_streak FROM economy WHERE member_id = ?", member.id)
        .fetch_optional(&ctx.data().db_pool)
        .await?;

    if let Some(eco) = economy {
        streak_bonus = (eco.daily_streak * 50).min(500);
    }

    let total_reward = base_reward + streak_bonus;

    sqlx::query!(
        "UPDATE members SET coins = coins + ?, last_daily = CURRENT_TIMESTAMP WHERE id = ?",
        total_reward,
        member.id
    )
    .execute(&ctx.data().db_pool)
    .await?;

    sqlx::query!(
        "UPDATE economy SET total_earned = total_earned + ?, daily_streak = daily_streak + 1 WHERE member_id = ?",
        total_reward,
        member.id
    )
    .execute(&ctx.data().db_pool)
    .await?;

    let mut embed = success_embed(&format!("You received your daily reward of **{}** coins", total_reward));
    if streak_bonus > 0 {
        embed = embed.field("🔥 Streak Bonus", format!("{} coins", streak_bonus), false);
    }

    ctx.send(poise::CreateReply::default().embed(embed)).await?;
    Ok(())
}

#[poise::command(slash_command, category = "Economy")]
pub async fn shop(ctx: Context<'_>) -> Result<(), Error> {
    let embed = serenity::CreateEmbed::default()
        .title("🛒 Shop")
        .description("Use `/buy <item>` to purchase an item")
        .field("⭐ VIP Pass", "VIP role for 30 days\n**Price:** 10,000 💰\n**ID:** `vip`", false)
        .field("🎨 Custom Color", "Role with a custom color\n**Price:** 5,000 💰\n**ID:** `color`", false)
        .field("⚡ XP Boost x2", "Double XP for 24 hours\n**Price:** 3,000 💰\n**ID:** `boost`", false)
        .field("💰 Coin Pack", "Bonus 1,000 coins\n**Price:** 1,000 💰\n**ID:** `coins`", false)
        .color(serenity::Color::from_rgb(52, 152, 219));

    ctx.send(poise::CreateReply::default().embed(embed)).await?;
    Ok(())
}

#[poise::command(slash_command, category = "Economy")]
pub async fn buy(
    ctx: Context<'_>,
    #[description = "Item ID to buy"] item: String,
) -> Result<(), Error> {
    let guild_id = ctx.guild_id().ok_or("This command can only be used in a guild")?;
    let guild_id_db = guild_id.get() as i64;
    let user_id_db = ctx.author().id.get() as i64;

    let member = database::get_or_create_member(&ctx.data().db_pool, guild_id_db, user_id_db, &ctx.author().name).await?;

    let (item_name, price) = match item.as_str() {
        "vip" => ("VIP Pass", 10000),
        "color" => ("Custom Color", 5000),
        "boost" => ("XP Boost x2", 3000),
        "coins" => ("Coin Pack", 1000),
        _ => {
            ctx.send(poise::CreateReply::default().embed(error_embed("Item not found in the shop")))
                .await?;
            return Ok(());
        }
    };

    if member.coins < price {
        ctx.send(poise::CreateReply::default().embed(error_embed(&format!(
            "You do not have enough coins. Cost: {} 💰",
            format_number(price as i64)
        )))).await?;
        return Ok(());
    }

    sqlx::query!(
        "UPDATE members SET coins = coins - ? WHERE id = ?",
        price,
        member.id
    )
    .execute(&ctx.data().db_pool)
    .await?;

    let economy = sqlx::query!("SELECT inventory FROM economy WHERE member_id = ?", member.id)
        .fetch_optional(&ctx.data().db_pool)
        .await?;

    if let Some(eco) = economy {
        let inventory_str = eco.inventory.unwrap_or_else(|| "{}".to_string());
        let mut inventory: serde_json::Value = serde_json::from_str(&inventory_str).unwrap_or(json!({}));

        if let Some(obj) = inventory.as_object_mut() {
            let count = obj.get(&item).and_then(|v| v.as_i64()).unwrap_or(0);
            obj.insert(item.clone(), json!(count + 1));
        }

        let inventory_str = inventory.to_string();

        sqlx::query!(
            "UPDATE economy SET inventory = ?, total_spent = total_spent + ? WHERE member_id = ?",
            inventory_str,
            price,
            member.id
        )
        .execute(&ctx.data().db_pool)
        .await?;
    }

    let embed = success_embed(&format!("You purchased **{}**", item_name))
        .field("Price", format!("{} 💰", format_number(price as i64)), true)
        .field("Remaining balance", format!("{} 💰", format_number((member.coins - price) as i64)), true);

    ctx.send(poise::CreateReply::default().embed(embed)).await?;
    Ok(())
}

#[poise::command(slash_command, category = "Economy")]
pub async fn inventory(
    ctx: Context<'_>,
    #[description = "User to check inventory"] user: Option<serenity::User>,
) -> Result<(), Error> {
    let user = user.as_ref().unwrap_or_else(|| ctx.author());
    let guild_id = ctx.guild_id().ok_or("This command can only be used in a guild")?;

    let guild_id_db = guild_id.get() as i64;
    let user_id_db = user.id.get() as i64;

    let member = database::get_or_create_member(&ctx.data().db_pool, guild_id_db, user_id_db, &user.name).await?;

    let economy = sqlx::query!("SELECT inventory FROM economy WHERE member_id = ?", member.id)
        .fetch_optional(&ctx.data().db_pool)
        .await?;

    let mut embed = serenity::CreateEmbed::default()
        .title(format!("🎒 Inventory for {}", user.name))
        .thumbnail(user.face())
        .color(serenity::Color::from_rgb(155, 89, 182));

    if let Some(eco) = economy {
        if let Some(inv_str) = eco.inventory {
            if let Ok(inventory) = serde_json::from_str::<serde_json::Value>(&inv_str) {
                if let Some(obj) = inventory.as_object() {
                    if obj.is_empty() {
                        embed = embed.description("Inventory empty");
                    } else {
                        for (item_id, quantity) in obj {
                            let item_name = match item_id.as_str() {
                                "vip" => "⭐ VIP Pass",
                                "color" => "🎨 Custom Color",
                                "boost" => "⚡ XP Boost x2",
                                "coins" => "💰 Coin Pack",
                                _ => item_id,
                            };
                            embed = embed.field(item_name, format!("Quantity: {}", quantity), true);
                        }
                    }
                } else {
                    embed = embed.description("Inventory empty");
                }
            } else {
                embed = embed.description("Inventory empty");
            }
        } else {
            embed = embed.description("Inventory empty");
        }
    } else {
        embed = embed.description("Inventory empty");
    }

    ctx.send(poise::CreateReply::default().embed(embed)).await?;
    Ok(())
}

#[poise::command(slash_command, category = "Economy")]
pub async fn transfer(
    ctx: Context<'_>,
    #[description = "User to transfer to"] user: serenity::User,
    #[description = "Amount to transfer"] amount: i64,
) -> Result<(), Error> {
    if user.bot {
        ctx.send(poise::CreateReply::default().embed(error_embed("You cannot transfer coins to a bot")))
            .await?;
        return Ok(());
    }

    if user.id == ctx.author().id {
        ctx.send(poise::CreateReply::default().embed(error_embed("You cannot transfer coins to yourself")))
            .await?;
        return Ok(());
    }

    if amount <= 0 {
        ctx.send(poise::CreateReply::default().embed(error_embed("Amount must be greater than 0")))
            .await?;
        return Ok(());
    }

    let guild_id = ctx.guild_id().ok_or("This command can only be used in a guild")?;
    let guild_id_db = guild_id.get() as i64;
    let author_id_db = ctx.author().id.get() as i64;
    let user_id_db = user.id.get() as i64;

    let sender = database::get_or_create_member(&ctx.data().db_pool, guild_id_db, author_id_db, &ctx.author().name).await?;

    if sender.coins < amount {
        ctx.send(poise::CreateReply::default().embed(error_embed(&format!(
            "You do not have enough coins. Balance: {} 💰",
            format_number(sender.coins as i64)
        )))).await?;
        return Ok(());
    }

    let receiver = database::get_or_create_member(&ctx.data().db_pool, guild_id_db, user_id_db, &user.name).await?;

    sqlx::query!(
        "UPDATE members SET coins = coins - ? WHERE id = ?",
        amount,
        sender.id
    )
    .execute(&ctx.data().db_pool)
    .await?;

    sqlx::query!(
        "UPDATE members SET coins = coins + ? WHERE id = ?",
        amount,
        receiver.id
    )
    .execute(&ctx.data().db_pool)
    .await?;

    let embed = success_embed(&format!("You transferred {} 💰 to {}", format_number(amount), user.name))
        .field("Your new balance", format!("{} 💰", format_number(sender.coins - amount)), true);

    ctx.send(poise::CreateReply::default().embed(embed)).await?;
    Ok(())
}

#[poise::command(slash_command, category = "Economy")]
pub async fn gamble(
    ctx: Context<'_>,
    #[description = "Amount to gamble"] amount: i64,
) -> Result<(), Error> {
    if amount <= 0 {
        ctx.send(poise::CreateReply::default().embed(error_embed("Amount must be greater than 0")))
            .await?;
        return Ok(());
    }

    let guild_id = ctx.guild_id().ok_or("This command can only be used in a guild")?;
    let guild_id_db = guild_id.get() as i64;
    let user_id_db = ctx.author().id.get() as i64;

    let member = database::get_or_create_member(&ctx.data().db_pool, guild_id_db, user_id_db, &ctx.author().name).await?;

    if member.coins < amount {
        ctx.send(poise::CreateReply::default().embed(error_embed(&format!(
            "You do not have enough coins. Balance: {} 💰",
            format_number(member.coins as i64)
        )))).await?;
        return Ok(());
    }

    let win = rand::random::<bool>();

    let embed = if win {
        sqlx::query!(
            "UPDATE members SET coins = coins + ? WHERE id = ?",
            amount,
            member.id
        )
        .execute(&ctx.data().db_pool)
        .await?;

        serenity::CreateEmbed::default()
            .title("🎰 You won!")
            .description(format!("You gambled **{}** coins and won **{}** coins", format_number(amount), format_number(amount)))
            .field("New balance", format!("{} 💰", format_number(member.coins + amount)), true)
            .color(serenity::Color::from_rgb(46, 204, 113))
    } else {
        sqlx::query!(
            "UPDATE members SET coins = coins - ? WHERE id = ?",
            amount,
            member.id
        )
        .execute(&ctx.data().db_pool)
        .await?;

        serenity::CreateEmbed::default()
            .title("🎰 You lost!")
            .description(format!("You gambled **{}** coins and lost everything", format_number(amount)))
            .field("New balance", format!("{} 💰", format_number(member.coins - amount)), true)
            .color(serenity::Color::from_rgb(231, 76, 60))
    };

    ctx.send(poise::CreateReply::default().embed(embed)).await?;
    Ok(())
}
